src/translate.py

```diff
@@ -4169,8 +4190,8 @@ def process_instr(instr: Instruction) -> None:
             # if source = target, maybe we could overwrite that variable instead
             # of creating a new one?
             set_reg(target, val)
-            mn_parts = mnemonic.split(".")
-            if mnemonic.endswith("."):
+            mn_parts = arch_mnemonic.split(".")
+            if arch_mnemonic.startswith("ppc:") and arch_mnemonic.endswith("."):
```

Comment:
(a bit weird that we're doing `.rstrip(".")` for all archs, but only giving the instruction special behavior for ppc. I guess making that strictly correct just bloats the code unnecessarily much though)

Reply:
Yeah agreed - I was kind of walking a line here between brevity & correctness, in particular because this check is repeated a few times throughout translate.py. (I felt it was worth adding `arch_mnemonic.startswith("ppc:") ` as long as it could still fit on one line)